xapian-core/include/xapian/cluster.h

```diff
@@ -219,8 +219,8 @@ class Centroid : public PointType {
     /// Constructor
     Centroid() { magnitude = 0; }
 
-    /// This method initializes the values of a centroid to the Point 'x'
-    void set_to_point(Point &x);
+    // Constructor with Point argument
+    Centroid(Point &x);
```

Comment:
This makes the code slightly more compact to look at, but I feel its more intuitive to have the _set_to_point()_ function.

> It feels like having Centroid& c = p; work might be unexpected.

I don't think having this work would be unexpected because the centroid and point share almost the same interface and data members. They differ only in the way they are interacted with, and so the explicit constructor should be fine. I'm still a little biased towards having something like `set_to_point(Point& x)` since easier to understand.